src/components/flow/actions/sendbroadcast/SendBroadcastForm.tsx

```diff
@@ -242,7 +201,7 @@ export default class SendBroadcastForm extends React.Component<
 
       const templateVariables =
         this.state.templateVariables.length === 0 ||
-        (this.state.template.value && this.state.template.value.id !== template.id)
+        (this.state.template.value && this.state.template.value.uuid !== template.uuid)
           ? range(0, templateTranslation.variable_count).map(() => {
```

Comment:
@pankaj-ag The main issue was the check with id in these lines. Floweditor uses template ids and we use uuids.
We needed to check based on uuid of the template rather than the id to update variables that we are sending